tools/containerize_gcsfuse_docker/Dockerfile

```diff
@@ -43,21 +43,21 @@ ENV GOOS=linux
 ENV GO111MODULE=auto
 
 ARG GCSFUSE_VERSION
-ARG GCSFUSE_REPO="github.com/googlecloudplatform/gcsfuse/"
-ENV GCSFUSE_PATH "$GOPATH/src/$GCSFUSE_REPO"
-RUN go get -d ${GCSFUSE_REPO}
+ARG GCSFUSE_REPO="https://github.com/googlecloudplatform/gcsfuse/"
+ARG BRANCH_NAME="v${GCSFUSE_VERSION}"
+RUN git clone --branch ${BRANCH_NAME} ${GCSFUSE_REPO}
```

Comment:
I think, we should confirm - does it work with commitId or tag?

Reply:
@raj-prince  Apparently, it works for a branch or a tag, but not for a given commit id/SHA. There is no way to clone at a specific commit. 

From `git help clone`

```
-b <name>, --branch <name>
           Instead of pointing the newly created HEAD to the branch pointed to by the cloned repository’s HEAD, point to <name> branch instead. In a non-bare repository, this is the branch that will be checked out.  --branch can also take tags and detaches the HEAD at that
           commit in the resulting repository.
```

I tested this locally too.

I also found out and inserted the `--depth 1` flag in the command to reduce the `git clone` runtime to fetch only the state of the repo at the HEAD commit, skipping fetching any history.